pycbc/psd/variation.py

```diff
@@ -160,3 +165,220 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+
+    # Create a bandpass filter between low_freq and high_freq
+    filt = sig.firwin(4*srate, [low_freq, high_freq], pass_zero=False,
+                      window='hann', nyq=srate/2)
+    filt.resize(int(psd_duration*srate))
+    # Fourier transform the filter and take the absolute value to get
+    # rid of the phase. Save the filter as a frequency series.
+    filt = abs(rfft(filt))
+    my_filter = FrequencySeries(filt, delta_f=1./psd_duration,
+                                dtype=fs_dtype)
+
+    ind = 0
+    for tlong in times_long:
+        # Calculate PSD for long segment
+        if tlong + psd_long_segment <= float(end_time):
+            astrain = strain.time_slice(tlong, tlong + psd_long_segment)
+            plong = pycbc.psd.welch(astrain,
+                                    seg_len=int(psd_duration
+                                                * strain.sample_rate),
+                                    seg_stride=int(psd_stride
+                                                   * strain.sample_rate),
+                                    avg_method=psd_avg_method)
+        else:
+            astrain = strain.time_slice(tlong, end_time)
+            plong = pycbc.psd.welch(
+                           strain.time_slice(end_time - psd_long_segment,
+                                             end_time),
+                           seg_len=int(psd_duration * strain.sample_rate),
+                           seg_stride=int(psd_stride * strain.sample_rate),
+                           avg_method=psd_avg_method)
+
+        # Make the weighting filter - bandpass, which weight by f^-7/6,
+        # and whithen. The normalization is chosen so that the variance
+        # will be one if this filter is applied to white noise which
+        # already has a variance of one.
+        freqs = FrequencySeries(plong.sample_frequencies,
+                                delta_f=plong.delta_f,
+                                epoch=plong.epoch, dtype=fs_dtype)
+        fweight = freqs**(-7./6.) * my_filter / numpy.sqrt(plong)
+        fweight[0] = 0.
+        norm = 1. / numpy.sqrt(sum(abs(fweight)**2) / (len(fweight)-1.))
+        fweight = norm * fweight
+        fwhiten = numpy.sqrt(2./srate) / numpy.sqrt(plong)
+        fwhiten[0] = 0.
+        full_filt = sig.hann(int(psd_duration*srate))*numpy.roll(
+            irfft(fwhiten*fweight), int(psd_duration/2)*srate)
+        # Convolve the filter with long segment of data
+        wstrain = TimeSeries(sig.fftconvolve(astrain, full_filt,
+                                             mode='same'),
+                             delta_t=strain.delta_t,
+                             epoch=astrain.start_time)[int(strain_crop
+                                                           * srate):
+                                                       -int(strain_crop
+                                                            * srate)]
```

Comment:
hard to read/parse this very long line : split into 2, in the first line calculate wstrain without the indexing, on the second do `wstrain = wstrain[int(strain_crop * srate) : -int(strain_crop * srate)]`